### PR TITLE
[spaceship] add a check in app_version.setup_screenshots to also check if there is an app_preview already uploaded

### PR DESCRIPTION
**Pull request by xpy**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [ ] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
When trying to do any update, fastlane will first update the `scaled` flags of the display_families to `true` **unless** they have a screenshot. This doesn't take in account that a trailer (app preview) might be present instead. As a result, the `scaled` flag is set to `true` for said display_family and upon saving, the server complains that *Your app preview can't be uploaded because you have selected to use an app preview from a larger display size*.

This happens when a user tries to do a deliver, even with the `skip_screenshots` flags set to `true`. And that is because Fastlane will send the whole configuration again in order to update any part of it. The prerequisites are that there should be **only video previews** in a display_family that can be set as scalable. It might also happen if the `overwrite_screenshots` flag is also set to `true`.

### Description
There is one more check added to check if the `trailers` key is present and the `value` key of it has contents.

### CODE REVIEW

**Comment by xpy** [comment]
I signed it!

**Comment by janpio** [comment]
The comment at the beginning of the method might need an update as well.

Can you also explain the bigger picture here? What would a user do to trigger this code?

**Comment by max-ott** [comment]
Hey @xpy, looks like a great PR. What’s the fastest way to test this? Can I directly test it with spaceship?

**Comment by xpy** [comment]
@max-ott Hello! I have to admit that I only tested around my case and I didn't dive into other cases. What I do is `fastlane deliver --skip_binary_upload true --run_precheck_before_submit false --force true --overwrite_screenshots true` I also tried with `--skip_screenshots` and `--skip_metadata` and still have the same problem. And on the App Store Connect I have only one app preview video on the iPhone 5.5'', but I suspect it would also trigger for any display size that can be checked as "scalable".

**Comment by itchingpixels** [comment]
hi! we're encountering the same issue, is there any chance this will be merged? thank you!

**Comment by max-ott** [comment]
It needs more tests. We can’t simply merge each PR without enough tests and reviews from maintainers. As there were no other ways presented to easily test and debug this, its still waiting for more input from the author.

**Comment by xpy** [comment]
@max-ott I already mentioned that I tested for the very case that the PR is about. I also described the way to reproduce, if it seems difficult to reproduce, it probably is because it is indeed a very specific case that is difficult to reproduce. I am not sure what else you expect from me to do to validate this PR.

**Comment by max-ott** [comment]
@xpy as this changes `spaceship` behaviour it would be useful to have steps to reproduce via `spacecship`. PRs require tests from other people than the initial contributor and if no one else can test it, it won't get merged.

Easiest way would be to provide steps for the `spaceship` playground / IRB. (https://github.com/fastlane/fastlane/blob/master/spaceship/docs/AppStoreConnect.md)

**Comment by max-ott** [comment]
I've just had some time to review the code. I'm not sure if it's the right place to put the code there and it solves all issues. The issue at the moment with the app preview upload in general seems to be related to the `scaled` property.

We need to disabled the `scaled` property for the `display_family`, if:
- There are screenshots present (probably the case right now) 
- There are app previews present
- If you are trying to upload an app preview
- If you are trying to upload a screenshot (probably the case right now)

**Comment by mackoj** [comment]
This fix, fixes our CI has anyone have any issue with it ?

**Comment by rabbitinspace** [comment]
fixed #15036 for me as well

**Comment by mackoj** [comment]
Hi @joshdholtz,
Could you review this PR please ?

**Comment by joshdholtz** [comment]
Ah yes yes! I will do some review on this this week!  Thanks for friendly ping ❤️

**Comment by mackoj** [comment]
How could we help with this PR ? Did you get time to review it @joshdholtz ?

**Comment by joshdholtz** [comment]
@mackoj Thanks for ping on this! Got lost in the shuffle 😔 Bumping this to top of list for rest of this week

**Review by joshdholtz — APPROVED** [review]
This looks great! Sorry it took so long to get through 😔 I'm really trying to improve my personal processes to keep conversations and merges going. Sorry to everyone that needed this but really appreciate the contribution! ❤️

**Pull request merged**
